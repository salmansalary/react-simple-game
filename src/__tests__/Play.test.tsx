import React from 'react';
import { act } from 'react-dom/test-utils';

import { shallow, mount } from 'enzyme';
import Play from '../routes/play';

describe('Play', () => {
    
    window.HTMLMediaElement.prototype.load = () => { /* do nothing */ };
    window.HTMLMediaElement.prototype.play = () => { return {}  as any;/* do nothing */ };
    window.HTMLMediaElement.prototype.pause = () => { /* do nothing */ };

    it('should render App correctly', () => {
        const component = shallow(<Play initialGameTime={4} charStates = {{0:true}}/>);
    
        expect(component).toMatchSnapshot();
    });

    it('should score change after clicking on characters', async () => {

        let component;

        await act(async function() {
            component = mount(<Play initialGameTime={5} charStates = {{0:true}}/>);
    
            //Clicking on all the Characters to capture at lease one of the active Characters
            await new Promise(resolve => setTimeout(()=>resolve(true),800))

            component.find('div.character').map(char=>{
                char.simulate('click');
            })
        })
        
        expect(component.find('label.score').text()).not.toBe('0')
    });
});

