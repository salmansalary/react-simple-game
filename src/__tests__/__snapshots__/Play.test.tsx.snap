// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Play should render App correctly 1`] = `ShallowWrapper {}`;
